documentation/SolutionsRegistryFormat.md

```diff
@@ -143,6 +143,37 @@ This directive is used to detect whether a solution is installed. If any of thes
 ]
 ```
 
+####isConfigurable
+This is run before configuration to ensure that the application is actually ready to be configured. This is relevant for applications where e.g. a configuration file needs to be present, a wizard needs to be run on the first launch, etc. Each of the blocks will be evaluated, if *any* of them evaluates to true, the solution is considered configurable. The absence of an `isConfigurable` block is also interpreted as the solution being configurable.
+
+If a solution is not configurable, the `makeConfigurable` block will be executed immediately following the `isConfigurable` check (see below).
+
+**Example Entry**:
+```
+"isConfigurable": [
+    {
+        "type": "gpii.lifecycleActions.pathExists",
+        "path": "/tmp/fakemag1.settings.json"
+    }
+]
+```
+
+####makeConfigurable
+Is the actions that need to be taken to make the application configurable (such as running a wizard, creating a default configuration file, adding a new system user, etc). This will be triggered by the system if `isConfigurable` has evaluated to false and run immediately following the `isConfigurable` check.
+
+**Example Entry**:
+```
+"makeConfigurable": [
+    {
+        "type": "gpii.lifecycleActions.createFile",
```

Comment:
Yes, it sounds good - we may as well verify that this functionality is actually good for something, since it is no longer good for the original use case. Is it possible that the ORCA settings handler might be the only genuine use for this function in our current solutions? If we can demonstrate that we can simplify the logic there and have it simply consist of the JSON settings handler, then we have a reasonable win. If we can't actually find a single genuine use of this system, I wonder whether we shouldn't mothball it. I know it's an awful thing to suggest given you have put so much work into this pull, but I think it is desirable to keep the principle that any code we have in the codebase meets at least one real use case.